pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -123,11 +109,21 @@ func (n *NodeLabeller) execute() bool {
 	defer n.queue.Done(key)
 
 	if !n.kvmController.NodeHasDevice(kvmPath) {
-		n.logger.Errorf("node-labeller cannot work without KVM device.")
+		if !n.kvmPostedError {
+			n.logger.Errorf("node-labeller cannot work without KVM device.")
+			n.kvmPostedError = true
+		}
 		return true
 	}
 
-	err := n.run()
+	err := n.loadAll()
+	if err != nil {
+		n.logger.Errorf("node-labeller load error encountered: %v", err)
+		n.queue.AddRateLimited(key)
+		return true
+	}
```

Comment:
If this fails, why would it ever pass after being retried multiple times?

if this function is not eventually consistent, it needs to only be executed once at initialization, and a decision needs to be made at that point in time about how to handle a potential error.